Dockerfiles/gadget-default-debug.Dockerfile

```diff
@@ -0,0 +1,104 @@
+# Dockerfile for Inspektor Gadget container image.
+# This image contains CO-RE and BCC-based gadgets. Its base image is the
+# BCC image. It's the default image that is deployed in Inspektor Gadget.
+
+ARG BUILDER_IMAGE=debian:bullseye
+
+# BCC built from the gadget branch in the kinvolk/bcc fork.
+# See BCC section in docs/CONTRIBUTING.md for further details.
+ARG BCC="quay.io/kinvolk/bcc:b3b9bd12109fefce4dc31314cecdaeac5bf0d83e-focal-release"
+
+FROM ${BCC} as bcc
+FROM --platform=${BUILDPLATFORM} ${BUILDER_IMAGE} as builder
+
+ARG ENABLE_BTFGEN=false
+ENV ENABLE_BTFGEN=${ENABLE_BTFGEN}
+
+ARG TARGETARCH
+# We need a cross compiler and libraries for TARGETARCH due to CGO.
+RUN set -ex; \
+	export DEBIAN_FRONTEND=noninteractive; \
+	apt-get update && \
+	apt-get install -y gcc make ca-certificates git && \
+	echo 'deb http://deb.debian.org/debian bullseye-backports main' >> /etc/apt/sources.list && \
+	dpkg --add-architecture ${TARGETARCH} && \
+	apt-get update && \
+	apt-get install -y golang-1.18 libelf-dev:${TARGETARCH} \
+		pkg-config:${TARGETARCH} libseccomp-dev:${TARGETARCH} && \
+	ln -s /usr/lib/go-1.18/bin/go /bin/go && \
+	if [ ${TARGETARCH} = 'arm64' ]; then \
+		apt-get install -y gcc-aarch64-linux-gnu; \
+	fi
+
+# Download BTFHub files
+COPY ./tools /btf-tools
+RUN set -ex; mkdir -p /tmp/btfs && \
+	if [ "$ENABLE_BTFGEN" = true ]; then \
+		cd /btf-tools && \
+		./getbtfhub.sh; \
+	fi
+
+# Cache go modules so they won't be downloaded at each build
+COPY go.mod go.sum /gadget/
+RUN cd /gadget && go mod download
+
+# This COPY is limited by .dockerignore
+COPY ./ /gadget
+RUN cd /gadget/gadget-container && \
+	if [ ${TARGETARCH} = 'arm64' ]; then \
+		export CC=aarch64-linux-gnu-gcc; \
+	fi; \
+	DEBUG=1 make -j$(nproc) TARGET_ARCH=${TARGETARCH} gadget-container-deps
+
+# Execute BTFGen
+RUN set -ex; \
+	if [ "$ENABLE_BTFGEN" = true ]; then \
+		cd /gadget && \
+		make btfgen BPFTOOL=/tmp/btfhub/tools/bin/bpftool.x86_64 \
+			BTFHUB_ARCHIVE=/tmp/btfhub-archive/ OUTPUT=/tmp/btfs/ -j$(nproc); \
+	fi
+
+# Main gadget image
+
+FROM bcc
+
+# This will be used by dlv server.
+ENV DEBUG_PORT=4040
```

Comment:
Instead of creating a new Dockerfile for the container with the debug feature enabled, could you reuse the main Dockerfile but add a `ARG DEBUG=false` and have an option in the Makefile to turn DEBUG=true?